backend/video/views.py

```diff
@@ -237,33 +261,82 @@ def list_recent(request):
     # will always have one video associated with it.
     # In the future, if that constraint is removed then we might need to alter the logic.
 
-    # Get the N latest transcripts from the DB for the user
-    recent_transcripts = [
-        (transcript.video, transcript.updated_at)
-        for transcript in Transcript.objects.filter(user=request.user.id).order_by(
-            "-updated_at"
-        )[:count]
-    ]
-
-    # Get the date of the nth recently updated trancript from the above list
-    least_recently_updated_transcript_date = recent_transcripts[-1][1]
-
-    # Get the latest translations from the DB for the user which are updated after the nth recently updated transcript
-    recent_translations = [
-        (translation.transcript.video, translation.updated_at)
-        for translation in Translation.objects.filter(user=request.user.id)
-        .filter(updated_at__gt=least_recently_updated_transcript_date)
-        .select_related("transcript")
-        .order_by("-updated_at")
-    ]
+    try:
+        # Collect only those transcripts where the video is checked as audio only
+        if audio_only:
+            # Get the list of videos that are audio only
+            audio_only_videos = Video.objects.filter(audio_only=True)
```

Comment:
Heyy, apart from these couple of lines, rest of the logic is same, right?
So can we please make them common?

Reply:
I have refactored the `list_recent` endpoint.

Reply:
Thanks!

Reply:
Can I merge this PR then?

Reply:
Uhh, I think we can merge this once the corresponding frontend changes for endpoint renaming is done.
Can you please inform Pranav about these endpoint that needs to be renamed? I think it should be simple changes.
Once done, we will merge & deploy both the frontend and backend together.

After that, we can sync-up with Harsh on Monday to explain how we can implement this import audio feature on the frontend. Pranav will not be able to work on this as we discussed earlier, since he has mid-sem exams from next week and may not be able to dedicate much time.

Reply:
Right right, I have already informed @pranav2305 about the changes.